packit/source_git.py

```diff
@@ -536,19 +468,107 @@ def _rebase_patches(self, from_branch):
 
+    def _populate_distro_dir(self):
+        """Copy files used in the distro to package and test the software to .distro."""
+        # TODO(csomh): Check that the dist-git working directory is pristine,
+        # that is, it's not dirty and has nothing git clean -xd would remove.
+        # Error out, if it's not the case, we don't want to copy untracked content
+        # to .distro.
+        command = ["rsync", "--archive", "--delete"]
+        for exclude in ["*.patch", "sources", ".git*"]:
+            command += ["--filter", f"exclude {exclude}"]
+
+        command += [
+            str(self.dist_git.local_project.working_dir) + "/",
+            str(self.distro_dir),
+        ]
+
+        self.distro_dir.mkdir(parents=True)
+        run_command(command)
+
+    def _reset_gitignore(self):
+        reset_rules = textwrap.dedent(
+            """\
+            # Reset gitignore rules
+            !*
+            """
+        )
+        Path(self.distro_dir, ".gitignore").write_text(reset_rules)
+
+    def _configure_syncing(self):
+        """Populate source-git.yaml"""
+        package_config = {}
+        if self.upstream_lp.git_url:
+            package_config["upstream_project_url"] = self.upstream_lp.git_url
+        package_config.update(
+            {
+                "upstream_ref": self.upstream_ref,
+                "downstream_package_name": self.package_name,
+                "specfile_path": f"{self.DISTRO_DIR}{self.package_name}.spec",
+                "patch_generation_ignore_paths": [self.DISTRO_DIR],
+                "patch_generation_patch_id_digits": 0,
+                "sync_changelog": True,
+                "synced_files": [
+                    {
+                        "src": self.DISTRO_DIR,
+                        "dest": ".",
+                        "delete": True,
+                        "filters": [
+                            "protect .git*",
+                            "protect sources",
+                            "exclude source-git.yaml",
+                            "exclude .gitignore",
+                        ],
+                    }
+                ],
+            }
+        )
+        lookaside_sources = self._get_lookaside_sources()
+        if lookaside_sources:
+            package_config["sources"] = lookaside_sources
+
+        Path(self.distro_dir, "source-git.yaml").write_text(
+            yaml.dump(
+                package_config,
+                Dumper=SafeDumperWithoutAliases,
+                default_flow_style=False,
+                sort_keys=False,
+            )
+        )
+
+    def _adjust_specfile(self):
+        """Remove patches from the spec-file
+
+        They are added back when transforming source-git to dist-git.
+        """
+        spec = Specfile(
+            f"{self.distro_dir}/{self.package_name}.spec",
+            sources_dir=self.dist_git.local_project.working_dir,
+        )
+        spec.read_patch_comments()
```

Comment:
I'm wondering where/how we use the comments when the `spec` is defined locally in this method (so the object gets lost when we exit this method).